lib/streamlit/elements/time_widgets.py

```diff
@@ -353,7 +369,9 @@ def _time_input(
 
         time_input_proto = TimeInputProto()
         time_input_proto.label = label
-        time_input_proto.default = time.strftime(parsed_time, "%H:%M")
+        time_input_proto.default = (
+            "" if parsed_time is None else time.strftime(parsed_time, "%H:%M")
```

Comment:
Empty string here is used to indicate that the default value is empty or? Also probably something to consider to unify via the `optional` proto parameter. In that case, we just don't set `default` if `parsed_time` is `None`.

Reply:
Yes, empty string indicates empty value, I tried to avoid changing protobuf field, I am a bit worried about this. In number Input I did not have a choice, since I'd have to pick some number as an indicator, but here `""` empty string seems fine and not invasive.

Are you sure allowing null/undefined here is a good idea?